### ISSUE TITLE
Exclude dir not working

### ISSUE THREAD

**Issue by herzadinata** [air-verse/air#678]
I use:
air v1.61.1, built with Go go1.23.2

This is my `.air.toml` file, put in the project root:
![Screenshot 2024-11-04 at 09 58 32](https://github.com/user-attachments/assets/19a537bc-eee1-4272-b85b-4c2f26bfb508)

But the `node_modules` dir in the project root is still being watched:
![Screenshot 2024-11-04 at 09 57 20](https://github.com/user-attachments/assets/a1e1d575-4f54-4374-be46-7f885c0bcf48)

Expectation:
The `node_modules` should not be watched

**Comment by istyf** [comment]
Hmm, "works on my machine". Are you sure that it is actually using that config file? Does it fail as expected if you change the build command to some nonsense such as `go build -o ./tmp/main ./nosuchdirectory` ?

**Comment by herzadinata** [comment]
@istyf 

Oh sorry, a little misunderstanding, i am not using it to build golang binary file, rather to watch / hot-reload my golang development environment.

```.toml
...
[build]
  args_bin = []
  bin = "./tmp/main"
  cmd = "go build -o ./tmp/main ."
  exclude_dir = ["tmp", "node_modules"]
...
```

So that is for excluding directory on `build` command?  That's not what I want to do.

I want to watch the changes on my golang project but excluding that `node_modules` from being watched.
So I tried this

```
...
root = "."
tmp_dir = "tmp"
exclude_dir = ["tmp", "node_modules"]

[build]
  args_bin = []
...
```
That's still not excluding `node_modules` from being watched. Is it capable of doing that?

**Comment by istyf** [comment]
No, it is not specific to the build command. It specifies what directories should be excluded from watching for changes. 

What do you want to happen when air detects a file change? What is supposed to be reloaded and how?

**Comment by istyf** [comment]
Sorry, I didn’t notice that you moved exclude_dir out of the [build] section. That does not work. 

If you remove lines from the toml those settings will be set to their default values, which aren’t necessarily empty. You need to change it back to have any hope of getting it to work. 

If you do not want to build anything you need to set cmd to something that succeeds but does nothing. One example is here: https://github.com/air-verse/air/issues/632#issuecomment-2315259975

**Comment by herzadinata** [comment]
@istyf  I meant building for production, but building the binary file into `tmp_dir` for watching purpose is is fine.

Okay so I changed back the exclude_dir:
![Screenshot 2024-11-06 at 09 10 37](https://github.com/user-attachments/assets/be773c3f-8e0a-488a-9ada-9e8b99d4348b)

Then run air by also pointing the air.toml. file:
![Screenshot 2024-11-06 at 09 10 51](https://github.com/user-attachments/assets/b405a7f6-52a0-488d-9f2c-c3079b0ec36e)

But still the `node_modules` is being watched:
![Screenshot 2024-11-06 at 09 11 40](https://github.com/user-attachments/assets/245d3198-13b9-4c9a-a758-483e3fd15470)

I am expecting the `node_modules` should not be watched as I specified in the `exclude_dir` above.

**Comment by tylersmalley** [comment]
I am also experiencing this same issue. I am getting restarts for files defined within the `exclude_dir` setting.

```
root = "."
tmp_dir = "tmp"

[build]
cmd = "go build -o ./dist/apiserver ./cmd/apiserver"
bin = "./dist/apiserver --prompt-db=db.json"
include_ext = ["go", "tmpl", "txt", "json", "js", "css", "html"]
exclude_dir = ["data", "node_modules", "dist", "ext/vscode"]
stop_on_error = true
log = "apiserver.log"

[log]
time = true

[misc]
clean_on_exit = true
```

However, if I save a file within that directory I still get an "has changed" notice and subsequent restart.

```
[10:17:47] ext/vscode/package.nls.json has changed
```

**Comment by xingdongzhe** [comment]
Hey guys! I found this, when you defined some keys twice,  air will use default config
So we got this issue
@herzadinata I see that you config defined `delay` twice
![image](https://github.com/user-attachments/assets/ce5561be-c955-4eba-be68-327b7923b82d)

**Comment by herzadinata** [comment]
@xingdongzhe 

Nope. I only defined "bin" inside "[build]" once

![Screenshot 2024-12-26 at 14 49 50](https://github.com/user-attachments/assets/a846fcd1-db08-4dcf-b5cb-28febe1d7657)

**Comment by xingdongzhe** [comment]
@herzadinata 
I'm sorry for the bin exception, this is my local test, please ignore it. 
I checked your complete .air.toml file and found that `delay` is defined twice. 
This will cause air to parse the toml file error but air not report, and  the default configuration file will be used.
That's why `exculde_dir` not working.
![image](https://github.com/user-attachments/assets/bda4bef1-feed-4dac-946e-960de093c26d)

**Comment by herzadinata** [comment]
@xingdongzhe 
You're right! Duplicate fields is the cause. When I remove duplicate fields, the "exclude_dir" is now working as expected! Thank you so much! You're awesome!

**Comment by xingdongzhe** [comment]
Next time, It can check with `air -c .air.toml`, it will log if your `.air.toml` got errors, not like `air` just slience when and use default config. That will confuse us that why our .air.toml not work
